src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,835 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+                    if (entityRelationship.LinkingObject is not null)
+                    {
+                        multipleCreateStructure.ReferencingRelationships.Add(new Tuple<string, object?>(relationshipName, relationshipFieldValues) { });
+                        continue;
+                    }
+
+                    string targetEntityName = entityRelationship.TargetEntity;
+                    Dictionary<string, IValueNode?> columnDataInSourceBody = MultipleCreateOrderHelper.GetBackingColumnDataFromFields(context, entityName, sourceEntityFields, sqlMetadataProvider);
+                    IValueNode? targetNode = GraphQLUtils.GetFieldNodeForGivenFieldName(objectFieldNodes: sourceEntityFields, fieldName: relationshipName);
+
+                    // In this function call, nestingLevel parameter is set as 0 which might not be accurate.
+                    // However, it is irrelevant because nestingLevel is used only for logging error messages and we do not expect
+                    // any errors to occur here. All errors are expected to have caught during request validation.
+                    string referencingEntityName = MultipleCreateOrderHelper.GetReferencingEntityName(context: context,
+                                                                                                      sourceEntityName: entityName,
+                                                                                                      targetEntityName: targetEntityName,
+                                                                                                      relationshipName: relationshipName,
+                                                                                                      metadataProvider: sqlMetadataProvider,
+                                                                                                      nestingLevel: 0,
+                                                                                                      columnDataInSourceBody: columnDataInSourceBody,
+                                                                                                      targetNodeValue: targetNode);
+
+                    if (string.Equals(entityName, referencingEntityName, StringComparison.OrdinalIgnoreCase))
+                    {
+                        multipleCreateStructure.ReferencedRelationships.Add(new Tuple<string, object?>(relationshipName, relationshipFieldValues) { });
+                    }
+                    else
+                    {
+                        multipleCreateStructure.ReferencingRelationships.Add(new Tuple<string, object?>(relationshipName, relationshipFieldValues) { });
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Helper method to that looks at the input fields of a given entity and classifies, populates the fields belonging to current entity and linking entity.
```

Comment:
Need better wording for the description.
```suggestion
        /// Helper method which traverses the input fields for a given entity record and populates the fields/values into the appropriate data structures storing the field/values belonging to the current entity and the linking entity.
```